docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,699 @@ id: pillars-turbomodules
+
+### Android: Create the `build.gradle` file
+
+Next, create a `build.gradle` file in the `android` folder, with the following contents:
+
+```kotlin title="build.gradle"
+buildscript {
+  ext.safeExtGet = {prop, fallback ->
+    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+  }
+  repositories {
+    google()
+    gradlePluginPortal()
+  }
+  dependencies {
+    classpath("com.android.tools.build:gradle:7.1.1")
+  }
+}
+
+apply plugin: 'com.android.library'
+apply plugin: 'com.facebook.react'
+
+android {
+  compileSdkVersion safeExtGet('compileSdkVersion', 31)
+
+  defaultConfig {
+    minSdkVersion safeExtGet('minSdkVersion', 21)
+    targetSdkVersion safeExtGet('targetSdkVersion', 31)
+    buildConfigField("boolean", "IS_NEW_ARCHITECTURE_ENABLED", "true")
+  }
+}
+
+repositories {
+  maven {
+    // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
+    url "$projectDir/../node_modules/react-native/android"
+  }
+  mavenCentral()
+  google()
+}
+
+dependencies {
+  implementation 'com.facebook.react:react-native:+'
+}
+
+react {
+    jsRootDir = file("../js/")
+    libraryName = "RTNCalculator"
+    codegenJavaPackageName = "com.RTNCalculator"
+}
+```
+
+Of interest in the `build.gradle` file:
+
+- The `defaultConfig` block, within the `android` block, adds a `buildConfigField` to enable the New Architecture.
+- The `react` block configures the CodeGen process. For Android, we need to specify:
+  - the `jsRootDir`, which contains the relative path to the JavaScript specs
+  - the `libraryName` we will use to link the library in the app.
+  - the `codegenJavaPackageName` which corresponds to the name of the Java package we will use for the code generated by **CodeGen**.
+
+## 4. Native Code
+
+For the final step in getting your TurboModule ready to go, you'll need to write some native code to connect the JavaScript side to the native platforms. This process requires two main steps:
+
+- Run **CodeGen** to see what it generates.
+- Write your native code, implementing the generated interfaces.
+
+When developing a React Native app that uses a TurboMOdule, it is responsibility of the app to actually generate the code using **CodeGen**. However, when developing a TurboModule as a library, we need to reference the generated code, and it is therefore useful to see what the app will generate.
+
+As first step for both iOS and Android, this guide shows how to execute manually the scripts used by **CodeGen** to generate the required code. Further information on **CodeGen** can be found [here](/docs/pillars-codegen.md)
+
+:::caution
+The code generated by the **CodeGen** in this step should not be committed to the versioning system. React Native apps are able to generate the code when the app is built. This allows to avoid any ABI incompatibility and to ensure that a consistent version of the **CodeGen** is used.
```

Comment:
This was mentioned in the Fabric guide: https://github.com/facebook/react-native-website/blob/fd81aa82a7aceb9ff3b10d7b279a3de1055daf94/docs/the-new-architecture/pillars-fabric-components.md?plain=1#L305. Should that also be changed? I'm not really familiar with what "distributing prebuilts" means in this specific context.

Reply:
When a distributor builds a library and distribute it to the public so that users don't have to build it themselves, they are creating a prefab.

We have ABI issues when a prefab uses a version of a library that is used also by the app and the version used by the prefab is different from the one used by the app (and potentially incompatible). For example, if a TM uses ReactNative 0.68 and we create and distribute a pre-built version of it and it is used by an app that runs 0.69, we may have issues.

> Should that also be changed?

Yes, I will update it there as well.

Reply:
Updated to be similar to the Fabric guide, but I made a slight edit:

```
The code generated by **Codegen** in this step should not be committed to the versioning system. React Native apps are able to generate the code when the app is built. This allows an app to ensure that all libraries have code generated for the correct version of React Native.
```

I'm not sure if this matches the intent; but I felt "ensure that a consistent version of the **CodeGen** is used across all the libraries." was a bit ambiguous -- all the libraries in what context? I assumed it meant all of a given app's libraries.